jupyter/d2l-java/chapter_multilayer-perceptrons/underfit-overfit.ipynb

```diff
@@ -0,0 +1,981 @@
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%mavenRepo snapshots https://oss.sonatype.org/content/repositories/snapshots/\n",
+    "\n",
+    "%maven ai.djl:api:0.6.0-SNAPSHOT\n",
+    "%maven org.slf4j:slf4j-api:1.7.26\n",
+    "%maven org.slf4j:slf4j-simple:1.7.26\n",
+    "%maven net.java.dev.jna:jna:5.3.0\n",
+    "%maven ai.djl.mxnet:mxnet-engine:0.6.0-SNAPSHOT\n",
+    "%maven ai.djl.mxnet:mxnet-native-auto:1.7.0-a"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%loadFromPOM\n",
+    "<dependency>\n",
+    "    <groupId>tech.tablesaw</groupId>\n",
+    "    <artifactId>tablesaw-jsplot</artifactId>\n",
+    "    <version>0.30.4</version>\n",
+    "</dependency>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%load ../utils/plot-utils.ipynb\n",
+    "%load ../utils/DataPoints.java"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import ai.djl.Device;\n",
+    "import ai.djl.Model;\n",
+    "import ai.djl.ndarray.NDArray;\n",
+    "import ai.djl.ndarray.NDManager;\n",
+    "import ai.djl.ndarray.index.NDIndex;\n",
+    "import ai.djl.ndarray.types.DataType;\n",
+    "import ai.djl.ndarray.types.Shape;\n",
+    "import ai.djl.nn.Block;\n",
+    "import ai.djl.nn.ParameterList;\n",
+    "import ai.djl.nn.SequentialBlock;\n",
+    "import ai.djl.nn.core.Linear;\n",
+    "import ai.djl.training.DefaultTrainingConfig;\n",
+    "import ai.djl.training.Trainer;\n",
+    "import ai.djl.training.dataset.ArrayDataset;\n",
+    "import ai.djl.training.dataset.Batch;\n",
+    "import ai.djl.training.listener.TrainingListener;\n",
+    "import ai.djl.training.loss.Loss;\n",
+    "import ai.djl.training.optimizer.Optimizer;\n",
+    "import ai.djl.training.optimizer.learningrate.LearningRateTracker;\n",
+    "import ai.djl.util.RandomUtils;\n",
+    "import ai.djl.training.EasyTrain;\n",
+    "import org.apache.commons.lang3.ArrayUtils;"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Generating the Dataset\n",
+    "\n",
+    "First we need data. Given $x$, we will use the following cubic polynomial to generate the labels on training and test data:\n",
+    "\n",
+    "$$y = 5 + 1.2x - 3.4\\frac{x^2}{2!} + 5.6 \\frac{x^3}{3!} + \\epsilon \\text{ where }\n",
+    "\\epsilon \\sim \\mathcal{N}(0, 0.1).$$\n",
+    "\n",
+    "The noise term $\\epsilon$ obeys a normal distribution\n",
+    "with a mean of 0 and a standard deviation of 0.1.\n",
+    "We will synthesize 100 samples each for the training set and test set."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "// utility functions used\n",
+    "\n",
+    "public float facti(float x) {\n",
```

Comment:
There could be ways to simplify the part in here:
http://d2l.ai/chapter_multilayer-perceptrons/underfit-overfit.html
@zachgk not sure if you have any ops in mind we can do to make it simple. Maybe considering adding a shuffle op?

Reply:
We will need a shuffle op because numpy has it and we also use it in DataIterable. I am fine with this for now